### PR TITLE
Add new acquire token methods

### PR DESCRIPTION
**Pull request by antrix1989**

### CODE REVIEW

File: MSAL/src/MSALPublicClientApplication.m [hunk 1]
```diff
@@ -365,8 +367,23 @@ + (BOOL)cancelCurrentWebAuthSession
...
 }
 
-#pragma mark -
-#pragma mark acquireToken
+#pragma mark - Acquire Token
+
+- (void)acquireTokenWithParameters:(MSALInteractiveTokenParameters *)parameters
+                   completionBlock:(MSALCompletionBlock)completionBlock
+{
+    [self acquireTokenForScopes:parameters.scopes
```
Comment (oldalton) [inline MSAL/src/MSALPublicClientApplication.m]: nit: would it make sense to flip it? Have an internal method that always accepts parameters and all overloads can create a parameter object and pass it to that method?

File: MSAL/src/public/MSALInteractiveTokenParameters.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+/*!
+ Token parameters to be used in interactive flow.
+ */
+@interface MSALInteractiveTokenParameters : MSALTokenParameters
+
+/*!
+ A specific UI behavior for the interactive authentication flow.
+ */
+@property (nonatomic) MSALUIBehavior uiBehavior;
```
Comment (oldalton) [inline MSAL/src/public/MSALInteractiveTokenParameters.h]: nit: we agreed to rename UIBehavior to prompt in all MSALs. Can be taken as a separate item, just adding a note here.

File: MSAL/src/public/MSALSilentTokenParameters.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ @param scopes      Permissions you want included in the access token received
+                    in the result in the completionBlock. Not all scopes are
+                    gauranteed to be included in the access token returned.
+ @param account     An account object retrieved from the application object that the
+                    interactive authentication flow will be locked down to.
+ */
+- (instancetype)initWithScopes:(NSArray<NSString *> *)scopes
+                       account:(MSALAccount *)account NS_DESIGNATED_INITIALIZER;
+
+- (instancetype)initWithScopes:(NSArray<NSString *> *)scopes NS_UNAVAILABLE;
```
Comment (oldalton) [inline MSAL/src/public/MSALSilentTokenParameters.h]: 👍

File: MSAL/src/public/MSALInteractiveTokenParameters.h (hunk 2 above)
Comment (antrix1989) [reply MSAL/src/public/MSALInteractiveTokenParameters.h]: Sure, let's do it in separate pr.

File: MSAL/src/MSALPublicClientApplication.m (hunk 1 above)
Comment (antrix1989) [reply MSAL/src/MSALPublicClientApplication.m]: We will need to change it in many places (~13 overloads), but I don't see much benefit in doing this way. Let's discuss it and address in separate pr.

**Comment by antrix1989** [comment]
> Should we also update code inside sample app to showcase recommended usage of new APIs?

Sample app was already moved to the new API.

**Pull request merged**